### DISCUSSION TITLE
How to specify cmd+enter behavior for forms on macOS?

### DISCUSSION THREAD

**Discussion by MartyIX** [dotnet/maui#12702] [Q&A]
I have a form with two text entries and two buttons:

```
[text entry 1]  <- initial focus is here
[text entry 2]

[button1][button2]
```

`button1` is supposed to be the one that is submitted when user presses cmd+enter. Is there any way to implement such behavior?

Thanks!

edit: Response from Drastic Actions:

https://learn.microsoft.com/en-us/search/?scope=Xamarin.iOS%20API&terms=KeyEquivalent
I'm not sure if there is a direct MAUI way to set keyboard accelerators
But you should be able to get the platform view and set a KeyEquivalent to it 
That gives you an event when a user presses said combination

**Comment by xtuzy** [comment]
i google get these links:
1. https://developer.apple.com/documentation/uikit/mac_catalyst/handling_key_presses_made_on_a_physical_keyboard?language=objc
2. https://stackoverflow.com/questions/60820752/how-do-i-listen-to-key-presses-in-a-mac-catalyst-app

according to 1, you can override uiviewcontroller"s pressesbegin method. from https://github.com/dotnet/maui/issues/7174, i know maui's page use uiviewcontroller, so you need use custom uiviewcontroller to create page, i don't know how to do. i remember uiview also have this method, you can try it.

**Reply by MartyIX** [reply]
Well, can't I just set https://developer.apple.com/documentation/uikit/uibutton/role/primary for a button (`UIButton`) so that it's automatically registered as a primary button and cmd+enter works immediately?

I have been trying for a couple of hours and so far no success. Maybe I'm just wrong :)
